robots/gazebo_tank.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+// BoB robotics includes
+#include "../common/pose.h"
+#include "../common/stopwatch.h"
+#include "tank.h"
+
+// Third-party includes
+#include "../third_party/units.h"
+#include <utility> 
+namespace BoBRobotics {
+namespace Robots {
+using namespace units::literals;
+
+template<typename LengthUnit = units::length::millimeter_t,
+         typename AngleUnit = units::angle::degree_t>
+class GazeboTank
+  : public Tank
+{
+    using millimeter_t = units::length::millimeter_t;
+    using meters_per_second_t = units::velocity::meters_per_second_t;
+
+public:
+    GazeboTank(const meters_per_second_t maximumSpeed, const millimeter_t axisLength)
+      : m_MaximumSpeed(maximumSpeed)
+      , m_AxisLength(axisLength)
+    {}
```

Comment:
I'm guessing that the maximum speed is either set in the Gazebo plugin or there is no maximum speed, so I guess we don't need this here. I'm guessing this is just leftover cruft from the ``SimulatedTank`` class.

Reply:
The way it's currently implemented is that the absolute wheel speeds should be sent via gazebo transport messages. So this maximum speed is multiplied with the joystick values to obtain an absolute speed which is finally transmitted to the Gazebo plugin. 
However, the `axisLength` value is not used at all, it was leftover from the SimulatedTank class. I'll remove it in the next commit.